--- libusb-1.0.8/libusb/io.c	2010-04-21 23:44:30.000000000 +0200
+++ libusb-1.0.8/libusb/io.c.new	2011-10-07 14:10:23.644816174 +0200
@@ -36,6 +36,13 @@
 
 #include "libusbi.h"
 
+#ifndef TIMESPEC_TO_TIMEVAL
+#define TIMESPEC_TO_TIMEVAL(tv, ts) do {                                \
+        (tv)->tv_sec = (ts)->tv_sec;                                    \
+        (tv)->tv_usec = (suseconds_t)(ts)->tv_nsec / 1000;              \
+} while (/*CONSTCOND*/0)
+#endif
+
 /**
  * \page io Synchronous and asynchronous device I/O
  *
